mono/metadata/threads.c

```diff
@@ -6016,6 +6016,8 @@ thread_summary_state_to_str (MonoSummaryState state)
 		return "MONO_SUMMARY_EXAMINE";
 	case MONO_SUMMARY_MUTATE_SHARED:
 		return "MONO_SUMMARY_MUTATE_SHARED";
+	default:
+		g_assert_not_reached ();
```

Comment:
I suggest:

```
static MonoSummaryState
thread_summary_state_validate (MonoSummaryState state)
{
	switch (state) {
	case MONO_SUMMARY_EMPTY:
	case MONO_SUMMARY_EXPECT:
	case MONO_SUMMARY_IN_PROGRESS:
	case MONO_SUMMARY_EXAMINE:
	case MONO_SUMMARY_MUTATE_SHARED:
		return state;
	default:
		g_assert_not_reached ();
	}
}

static const char *
thread_summary_state_to_str (MonoSummaryState state)
{
	switch (thread_summary_state_validate (state)) {
	case MONO_SUMMARY_EMPTY:
		return "MONO_SUMMARY_EMPTY";
	case MONO_SUMMARY_EXPECT:
		return "MONO_SUMMARY_EXPECT";
	case MONO_SUMMARY_IN_PROGRESS:
		return "MONO_SUMMARY_IN_PROGRESS";
	case MONO_SUMMARY_EXAMINE:
		return "MONO_SUMMARY_EXAMINE";
	case MONO_SUMMARY_MUTATE_SHARED:
		return "MONO_SUMMARY_MUTATE_SHARED";
	}
}
```

Reply:
Sure, I just don't see this pattern used anywhere in existing code, and I can't tell why it'd be better.